fe/fe-core/src/main/java/com/starrocks/sql/optimizer/SubqueryUtils.java

```diff
@@ -19,12 +18,21 @@
 import com.starrocks.sql.optimizer.operator.scalar.ColumnRefOperator;
 import com.starrocks.sql.optimizer.operator.scalar.ConstantOperator;
 import com.starrocks.sql.optimizer.operator.scalar.ScalarOperator;
+import com.starrocks.sql.optimizer.rewrite.BaseScalarOperatorShuttle;
 
+import java.util.Collection;
 import java.util.List;
 import java.util.Map;
 
 public class SubqueryUtils {
 
+    public static final String EXIST_NON_EQ_PREDICATE = "Not support Non-EQ correlated predicate in correlated subquery";
+
+    public static final String NONE_CORRELATED_PREDICATE = "Not support without correlated predicate in correlated subquery";
```

Comment:
It should called non-correlated subquery if not containing correlated predicate. How to understand this error message?

Reply:
The old logic split predicate in subquery into two groups(correlated and uncorrelated).  when correlated group is empty will throw an exception. It seems in the check() method we had ensure it contains correlated predicate. @Seaven  do you know the reason why we check it again? 
```
PushDownApplyAggFilterRule.class
if (correlationPredicate.isEmpty()) {
            // If the correlation predicate doesn't appear in here,
            // it should not be a situation that we currently support.
            throw new SemanticException(SubqueryUtils.NONE_CORRELATED_PREDICATE);
        }
```

Reply:
The push down rule depends on correlation predicate, if don't support correlation predicate, we can't push down apply node